### ISSUE TITLE
Message **send_and_save** does not raise an error when receiving invalid data 

### ISSUE THREAD

**Issue by YuvHayun** [ecederstrand/exchangelib#1159]
**Describe the bug**
The **send_and_save** does not inform about unsuccessful attempts to send messages.
The following code doesn't catch any exceptions:
```python
def send_email(self, message: Message):
        account = self.get_account()
        message.account = account
        try:
            message.send_and_save()
        except Exception:
            print("could not send mail.")
```
which eventually leads to the program printing a success message.

**To Reproduce**
Try to create a message where the 'to_recipients' is not valid mail (say 'a') and run **send_and_save** on the massage

**Expected behavior**
Expected a failure message when trying to create the Message entity / sending the Message.

**Additional context**
Not sure if this is a bug or the desired design, thx in advance!

**Comment by ecederstrand** [comment]
Thanks for the report! I agree this looks like a bug. I'll try to recreate in a test case so we can get this fixed.

**Comment by YuvHayun** [comment]
Thx! let me know if you need any more information.

**Comment by ecederstrand** [comment]
I got around to testing this today. I tried to send a message to an invalid email, but I just get a success message as response.

I think Exchange does this because it just acknowledges that the email was sent. It could take a long time for the email server in the other end to respond that the account does not exist, is over quota, or a long list of other issues, so the feedback is not instant. Eventually, you may get an message in your inbox with an error message.

In conclusion, I don't think there's anything to change in exchangelib. You're asking too much of the API :-)

**Comment by YuvHayun** [comment]
> I got around to testing this today. I tried to send a message to an invalid email, but I just get a success message as response.
> 
> I think Exchange does this because it just acknowledges that the email was sent. It could take a long time for the email server in the other end to respond that the account does not exist, is over quota, or a long list of other issues, so the feedback is not instant. Eventually, you may get an message in your inbox with an error message.
> 
> Looking at the code path, if you send an email and the server does respond with an error, then that error is raised correctly in exchangelib.
> 
> In conclusion, I don't think there's anything to change in exchangelib. You're asking too much of the API :-)

Got it, thank you very much!!
